prqlc/prql-compiler/src/sql/std.sql.prql

```diff
@@ -93,8 +93,25 @@ let round = n_digits column -> s"ROUND({column:0}, {n_digits:0})"
 let as = `type` column -> s"CAST({column:0} AS {type:0})"
 
 # String functions
+# Clickhouse: https://clickhouse.com/docs/en/sql-reference/functions/string-functions
+# DuckDB: https://duckdb.org/docs/sql/functions/char
+# MySQL: https://dev.mysql.com/doc/refman/8.0/en/string-functions.html
+# Postgres: https://www.postgresql.org/docs/7.4/functions-string.html
+# SQLite: https://www.sqlite.org/lang_corefunc.html
+# MSSQL: https://learn.microsoft.com/en-us/sql/t-sql/functions/string-functions-transact-sql
+# BigQuery: https://cloud.google.com/bigquery/docs/reference/standard-sql/string_functions
+# Snowflake: https://docs.snowflake.com/en/sql-reference/functions-string
 let lower = column -> s"LOWER({column:0})"
 let upper = column -> s"UPPER({column:0})"
+let ltrim = column -> s"LTRIM({column:0})"
+let rtrim = column -> s"RTRIM({column:0})"
+let trim = column -> s"TRIM({column:0})"
+let length = column -> s"CHAR_LENGTH({column:0})"
+let substring = offset length column -> s"SUBSTRING({column:0}, {offset:0}, {length:0})"
+let replace = pattern replacement column -> s"REPLACE({column:0}, {pattern:0}, {replacement:0})"
+let starts_with = prefix column -> s"{column} LIKE CONCAT({prefix}, '%')"
+let contains = substr column -> s"{column} LIKE CONCAT('%', {substr}, '%')"
+let ends_with = suffix column -> s"{column} LIKE CONCAT('%', {suffix})"
```

Comment:
These are nice functions to have.

But I think these ones are likely to only work with dialects that use single quotes and `CONCAT`.

Instead, could we use an [f-string](https://prql-lang.org/book/reference/syntax/f-strings.html) to construct the pattern, and then I think it'll work for everything? (but would need to check)

Reply:
How do you mix f-strings and s-strings ?

Reply:
Ah good point, that would require https://github.com/PRQL/prql/pull/3280. Then we could have something like `s"{column LIKE f'%{suffix}'}"`.

Can we maybe only add to the dialects that support it? Or that's a lot of work? I would emphasize moving quickly, but also we should try not to have things that we know don't work...

We could also leave those ones commented out, and I could take a another go at #3280 (but not sure whether #3280 is easy or difficult)

Reply:
I would love to have #3280 implemented but I'm afraid it's a lot of work.
I took a look at all the the supported dialects, only SQLite doesn't support `CONCAT`

Reply:
Looks great! Thanks for filling those out